packages/components/src/ui/item-group/styles.ts

```diff
@@ -1,73 +1,58 @@
 /**
  * External dependencies
  */
-// Disable reason: Temporarily disable for existing usages
-// until we remove them as part of https://github.com/WordPress/gutenberg/issues/30503#deprecating-emotion-css
-// eslint-disable-next-line no-restricted-imports
-import { css } from '@emotion/css';
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
 
 /**
  * Internal dependencies
  */
 import { CONFIG } from '../../utils';
 import COLORS from '../../utils/colors-values';
-
-export const unstyledButton = css`
-	appearance: none;
-	border: 1px solid transparent;
-	cursor: pointer;
-	background: none;
-	text-align: left;
-
-	&:hover {
-		color: ${ COLORS.admin.theme };
-	}
-
-	&:focus {
-		background-color: transparent;
-		color: ${ COLORS.admin.theme };
-		border-color: ${ COLORS.admin.theme };
-		outline: 3px solid transparent;
-	}
-`;
-
-export const item = css`
-	width: 100%;
-	display: block;
-`;
-
-export const bordered = css`
-	border: 1px solid ${ CONFIG.surfaceBorderColor };
-`;
-
-export const separated = css`
-	> *:not( marquee ) {
-		border-bottom: 1px solid ${ CONFIG.surfaceBorderColor };
-	}
-
-	> *:last-child:not( :focus ) {
-		border-bottom-color: transparent;
-	}
-`;
-
-const borderRadius = CONFIG.controlBorderRadius;
-
-export const spacedAround = css`
-	border-radius: ${ borderRadius };
-`;
-
-export const rounded = css`
-	border-radius: ${ borderRadius };
-
-	> *:first-child {
-		border-top-left-radius: ${ borderRadius };
-		border-top-right-radius: ${ borderRadius };
-	}
-
-	> *:last-child {
-		border-bottom-left-radius: ${ borderRadius };
-		border-bottom-right-radius: ${ borderRadius };
-	}
+import type { ItemGroupProps, ItemProps } from './types';
+import type { ItemGroupContext } from './context';
+
+type ItemGroupViewProps = Required< ItemGroupProps >;
+type ItemViewProps = Required< ItemProps >;
+
+const renderBordered = ( { isBordered }: ItemGroupViewProps ) =>
+	isBordered &&
+	css`
+		border: 1px solid ${ CONFIG.surfaceBorderColor };
+	`;
+
+const renderSeparated = ( { isSeparated, isBordered }: ItemGroupViewProps ) =>
+	( isBordered || isSeparated ) &&
+	css`
+		> *:not( marquee ) {
+			border-bottom: 1px solid ${ CONFIG.surfaceBorderColor };
+		}
+
+		> *:last-child:not( :focus ) {
+			border-bottom-color: transparent;
+		}
+	`;
+
+const renderRounded = ( { isRounded }: ItemGroupViewProps ) =>
+	isRounded &&
+	css`
+		border-radius: ${ borderRadius };
+
+		> *:first-child {
```

Comment:
This rule seems to create a warning in Storybook:

```The pseudo class ":first-child" is potentially unsafe when doing server-side rendering. Try changing it to ":first-of-type"```

```suggestion
		> *:first-of-type {
```